common/component_interface_utils/include/component_interface_utils/rclcpp/service_server.hpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COMPONENT_INTERFACE_UTILS__RCLCPP__SERVICE_SERVER_HPP_
+#define COMPONENT_INTERFACE_UTILS__RCLCPP__SERVICE_SERVER_HPP_
+
+#include <rclcpp/rclcpp.hpp>
+
+namespace component_interface_utils
+{
+
+template <class SpecT>
+class Service
+{
+public:
+  RCLCPP_SMART_PTR_DEFINITIONS(Service)
+
+  template <class NodeT>
+  using CallbackType = void (NodeT::*)(
+    typename SpecT::Service::Request::SharedPtr, typename SpecT::Service::Response::SharedPtr);
+
+  explicit Service(typename rclcpp::Service<typename SpecT::Service>::SharedPtr service)
+  {
+    service_ = service;
+  }
+
+  template <class CallbackT>
+  static auto wrap(CallbackT && callback, const rclcpp::Logger & logger)
+  {
+    auto wrapped = [logger, callback](
+                     typename SpecT::Service::Request::SharedPtr request,
+                     typename SpecT::Service::Response::SharedPtr response) {
+      using rosidl_generator_traits::to_yaml;
+      RCLCPP_INFO_STREAM(logger, "service call: " << SpecT::name << "\n" << to_yaml(*request));
+      callback(request, response);
+      RCLCPP_INFO_STREAM(logger, "service exit: " << SpecT::name << "\n" << to_yaml(*response));
```

Comment:
Don't you have to print `response`?

Reply:
Yes, I think there are cases where response status and return values are analyzed.

Reply:
Hmm? Will you add them or not?

Reply:
What does it mean? Response is already printed.

Reply:
Ah, I'm really sorry, I saw it by mistake. :bow: 
I thought the both are `request`. :sweat_smile: